### ISSUE TITLE
zwj emojis like hackercat don't validate as icon

### ISSUE THREAD

**Issue by jrabbit** [microsoft/terminal#8583]
I can't set my icon to 🐱‍💻 but 🤖 works as expected. in terminal-preview.

# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Windows Terminal version (if applicable):
 Win32NT             10.0.19042.0 Microsoft Windows NT 10.0.19042.0
Any other software?
```

# Steps to reproduce

Set Icon property to a fancy zwj emoji.

# Expected behavior

 Icon should be the emoji.
# Actual behavior
json doesn't validate

**Comment by skyline75489** [comment]
Yeah you're right. Right now the restriction for emoji strings are very strict. See:

https://github.com/microsoft/terminal/blob/b80a4e45cce0a9cb47e5bb1ad51f86eb84e59471/src/cascadia/TerminalSettingsModel/IconPathConverter.cpp#L114

We need a better way to detect complicated emojis.

**Comment by zadjii-msft** [comment]
Yep that's a good point. I wonder if ZWJ emoji will actually work as a `FonIcon` 😬 

Oh, no they won't
![image](https://user-images.githubusercontent.com/18356694/102077717-5a73df80-3dcf-11eb-9d73-57637302f62c.png)


So we'll need to file this upstream on WinUI

**Comment by skyline75489** [comment]
So ZWJ emojis can be very long and take up to about 8 wchars, according to [emoji-zwj-sequences](https://unicode.org/emoji/charts/emoji-zwj-sequences.html). Guess length check isn't going to suffice. I wonder if this can be an early adopter of the Unicode handling work by @reli-msft .

**Comment by fitojb** [comment]
The Windows 11 emoji set no longer supports the ninja cat sequences, so this issue may be considered moot now

**Comment by DHowett** [comment]
Nah, we should still make sure we support these. Important emoji like "Woman whose skin is not yellow" are ZWJ-joined as well. :smile:
